src/main/java/nextstep/subway/line/domain/Sections.java

```diff
@@ -0,0 +1,181 @@
+        }
+        return stations;
+    }
+
+    private void update(Section newSection) {
+        sections.forEach(section -> section.update(newSection));
+    }
+
+    public void deleteSection(Station station) {
+        validateNumberOfSections();
+        if (isEqualToFirstUpStation(station)) {
+            sections.remove(findSectionWithSameUpStation(station));
+            return;
+        }
+        if (isEqualToLastDownStation(station)) {
+            sections.remove(findSectionWithSameDownStation(station));
+            return;
+        }
+        Section nextSection = findSectionWithSameUpStation(station);
+        Section prevSection = findSectionWithSameDownStation(station);
+        sections.remove(nextSection);
+        sections.remove(prevSection);
+        sections.add(mergeSection(prevSection, nextSection));
+    }
+
+    private void validateNumberOfSections() {
+        if (!isSizeMoreThanMin()) {
+            throw new ImpossibleDeleteException("제거 가능한 구간이 없습니다.");
+        }
+    }
+
+    private boolean isSizeMoreThanMin() {
+        return sections.size() > MIN;
+    }
+
+    private boolean isEqualToFirstUpStation(Station station) {
+        return station.equals(findUpStation());
+    }
+
+    private Station findUpStation() {
+        Set<Station> upStations = getUpStations();
+        Set<Station> downStations = getDownStations();
+        return upStations.stream()
+                .filter(upStation -> !downStations.contains(upStation))
+                .findFirst()
+                .orElseThrow(() -> new NotFoundException("상행 종점역을 찾을 수 없습니다."));
+    }
+
+    private boolean isEqualToLastDownStation(Station station) {
+        return station.equals(findDownStation());
+    }
+
+    private Station findDownStation() {
+        Set<Station> upStations = getUpStations();
+        Set<Station> downStations = getDownStations();
+        return downStations.stream()
+                .filter(downStation -> !upStations.contains(downStation))
+                .findFirst()
+                .orElseThrow(() -> new NotFoundException("하행 종점역을 찾을 수 없습니다."));
+    }
+
+    private Set<Station> getUpStations() {
+        return sections.stream()
+                .map(Section::upStation)
+                .collect(Collectors.toSet());
+    }
+
+    private Set<Station> getDownStations() {
+        return sections.stream()
+                .map(Section::downStation)
+                .collect(Collectors.toSet());
+    }
+
+    private Section findSectionWithSameUpStation(Station station) {
+        return sections.stream()
+                .filter(section -> station.equals(section.upStation()))
+                .findFirst()
+                .orElseThrow(() -> new NotFoundException("일치하는 상행역이 없습니다."));
+    }
+
+    private Section findSectionWithSameDownStation(Station station) {
+        return sections.stream()
+                .filter(section -> station.equals(section.downStation()))
```

Comment:
```suggestion
               .filter(section -> section.sameDownStation(station))
```

비교하도록 시킨다면, 객체에게 데이터를 가져오지 않아도 되지 않을까요?
개인적으로 `getter`를 지양하는 것은 `get-`을 붙이지 않는것도 있겠지만, 데이터를 외부에서 가져다 쓰는것을 최소한으로 하라는것도 있는것 같아요 :thinking: 

이부분에 대해서 용일님 생각이 궁금하네요! :)